core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -737,8 +774,16 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
         removeFlag(DiplomacyFlags.BorderConflict)
 
 
-        // Go through city state allies.
         if (!civInfo.isCityState()) {
+            // Go through defensive pact allies.
+            if (!isOffensiveWar) {
+                for (defensivePact in civInfo.diplomacy.values.filter { it.diplomaticStatus == DiplomaticStatus.DefensivePact && !it.otherCiv().isDefeated()
+                    && !otherCivDiplomacy().civInfo.isAtWarWith(it.otherCiv())}) {
```

Comment:
This check doesn't make sense to me on first trading, it takes a while to understand what's going on, which to me indicates that readability is an issue. We need to call the parties by meaningful names